addons/account/static/src/components/char_with_placeholder_field/char_with_placeholder_field.xml

```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates xml:space="preserve">
+
+    <t t-name="account.CharWithPlaceholderField" t-inherit="web.CharField">
+        <xpath expr="//span" position="attributes">
+            <attribute name="t-att-class">{'text-muted': !this.props.record.data[props.name]}</attribute>
+        </xpath>
+        <xpath expr="//input" position="attributes">
+            <attribute name="t-att-placeholder">props.record.data[props.placeholderField] || props.placeholder</attribute>
```

Comment:
Can't you make `placeholder` be a getter for this expression instead of managing this in the view?